lhotse/augmentation/torchaudio.py

```diff
@@ -366,6 +368,92 @@ def reverse_timestamps(
         return offset, duration
 
 
+@dataclass
+class ReverbWithImpulseResponse(AudioTransform):
+    """
+    Reverberation effect by convolving with a room impulse response.
+    This code is based on Kaldi's wav-reverberate utility:
+    https://github.com/kaldi-asr/kaldi/blob/master/src/featbin/wav-reverberate.cc
+
+    The impulse response can possibly be multi-channel, in which case the reverberated audio
+    will be multi-channel as well.
+    """
+
+    rir: dict
+    shift_output: bool = True
+    normalize_output: bool = True
+
+    RIR_SCALING_FACTOR: float = 0.5 ** 15
+
+    def __call__(
+        self,
+        samples: np.ndarray,
+        sampling_rate: int,
+    ) -> np.ndarray:
+        """
+        :param samples: The audio samples to reverberate (must be single-channel).
+        :param sampling_rate: The sampling rate of the audio samples.
+        """
+        assert samples.shape[0] == 1, "The input audio must be single-channel."
+        sampling_rate = int(sampling_rate)  # paranoia mode
+
+        from lhotse import Recording
+
+        rir_ = Recording.from_dict(copy.deepcopy(self.rir)).load_audio()
```

Comment:
@pzelasko Here I am passing a copy of the rir dict so that the original doesn't get modified in the `from_dict()` method. Is this reasonable or is there a better way to do this?

Reply:
I see -- but I think that shallow copy should be sufficient here (from_dict is only popping keys, the values are untouched).

Reply:
That makes sense, thanks!